packages/components/table/src/util.ts

```diff
@@ -370,3 +370,116 @@ export function createTablePopper(
   on(trigger, 'mouseleave', removePopper)
   return popperInstance
 }
+
+export const isFixedColumn = <T>(
+  index: number,
+  fixed: string | boolean,
+  store: any,
+  realColumns?: TableColumnCtx<T>[]
+) => {
+  let start = 0
+  let after = index
+  if (realColumns) {
+    // handle group
+    for (let i = 0; i < index; i++) {
+      start += realColumns[i].colSpan
+    }
+    after = start + realColumns[index].colSpan - 1
+  } else {
+    start = index
+  }
+  let fixedLayout
+  const columns = store.states.columns
+  if (fixed === 'left') {
+    if (after < store.states.fixedLeafColumnsLength.value) {
+      fixedLayout = 'left'
+    }
+  } else if (fixed === 'right') {
+    if (
+      start >=
+      columns.value.length - store.states.rightFixedLeafColumnsLength.value
+    ) {
+      fixedLayout = 'right'
+    }
+  } else {
+    if (after < store.states.fixedLeafColumnsLength.value) {
+      fixedLayout = 'left'
+    } else if (
+      start >=
+      columns.value.length - store.states.rightFixedLeafColumnsLength.value
+    ) {
+      fixedLayout = 'right'
+    }
+  }
```

Comment:
**Suggestion**  Can we do a switch case on this? I think we should avoid 3 branches if-else block.